my_phase-retrieval/phase_retrieval.cu

```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <string>
+#include <cufft.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include <opencv2/imgcodecs.hpp>
+
+//store C++ and CUDA version of phase retrieval
+#include "phase_algo.hpp"
+#include "test_algo.hpp"
+
+namespace py = pybind11;
+
+PYBIND11_MODULE(cuPhaseRet, m) 
+{
+  //main phase retrieval
+  m.def("fienup_phase_retrieval", &fienup_phase_retrieval);
+
+  //test
```

Comment:
Please split up the python bindings in two header files. One for the actual algorithm and one for the tests.